freqtrade/optimize/backtesting.py

```diff
@@ -1070,6 +1077,26 @@ def backtest_one_strategy(self, strat: IStrategy, data: Dict[str, DataFrame],
         })
         self.all_results[self.strategy.get_strategy_name()] = results
 
+        if self.backtest_signal_candle_export_enable and \
+                self.dataprovider.runmode == RunMode.BACKTEST:
+            signal_candles_only = {}
+            for pair in preprocessed_tmp.keys():
+                signal_candles_only_df = DataFrame()
+
+                pairdf = preprocessed_tmp[pair]
+                resdf = results['results']
+                pairresults = resdf.loc[(resdf["pair"] == pair)]
+
+                if pairdf.shape[0] > 0:
+                    for t, v in pairresults.open_date.items():
+                        allinds = pairdf.loc[(pairdf['date'] < v)]
+                        signal_inds = allinds.iloc[[-1]]
+                        signal_candles_only_df = signal_candles_only_df.append(signal_inds)
```

Comment:
I wonder what the impact of this is ...

how much would this method take for a backtest of - say 1000 trades?
would that also cause the "fragmented dataframe" warning?

Reply:
1000 trades is only 1000 signal candles, so it's not that much of a problem IMO.